service/matching/taskReader.go

```diff
@@ -143,6 +143,9 @@ Loop:
 
 		case <-tr.notifyC:
 			tasks, readLevel, isReadBatchDone, err := tr.getTaskBatch()
+			if tr.tlMgr.errShouldUnload(err) {
+				tr.tlMgr.signalFatalProblem(tr.tlMgr)
+			}
```

Comment:
This check error and unload is repeated many times. We need a func to do this. Something like func maybeUnload(err).

Reply:
Added `taskQueueManagerImpl.signalIfFata(error) bool`